backend/src/api/extraction-job.js

```diff
@@ -0,0 +1,86 @@
+const { body, param } = require('express-validator');
+const multer = require('multer');
+const path = require('path');
+const process = require('process');
+const router = require('express').Router();
+
+const {
+  errorOnBadValidation,
+  fileUploadedValidator,
+  toMongooseIdSanitizer
+} = require('../utilities/custom-validation');
+const { ExtractionJob } = require('../database');
+const {
+  extractJobController,
+  ExtractionJobRequest,
+  FileMetadata
+} = require('../utilities/feature-extraction/extraction-job');
+const fileController = require('../utilities/file-controller');
+
+const DEFAULT_FILE_ENCODING = 'utf-8';
+const DEFAULT_USER_ID = 'Unknown user';
+const FILE_FIELD_IN_FORM = 'productFile';
+const UPLOAD_ROOT = path.join(__dirname, '../../resources/product-uploads'); // needs to be an absolute path
+
+const fileUpload = multer({ dest: UPLOAD_ROOT });
+
+/**
+ * Register listeners for upload cleanup. Uploaded files should be deleted when the job terminates,
+ * but if the user kills his/her API session early (or there is some issue with the DB), the uploaded file will not be cleaned up in the job.
+ * This is more of a 'double-check' than anything
+ */
+process.on('exit', () => fileController.deleteAllFilesInDirectory(UPLOAD_ROOT));
```

Comment:
Tried this on my machine, process doesn't fully exit here unless you invoke `process.exit(...)` in the handlers.

Reply:
Yeah, I ran into the same issue. That's what the 
`process.on('SIGINT', () => fileController.deleteAllFilesInDirectory(UPLOAD_ROOT));` is for.
The other is more of a "just in case" the user uses `process.exit(...)`